src/HypLSP7.sol

```diff
@@ -113,4 +115,42 @@ contract HypLSP7 is LSP7DigitalAssetInitAbstract, TokenRouter {
     {
         LSP7DigitalAssetInitAbstract._mint(_recipient, _amount, true, "");
     }
+
+    /**
+     * @inheritdoc TokenRouter
+     * @dev Ensures that the recipient of the token transfer is not the collateral contract
+     */
+    function transferRemote(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amountOrId
+    )
```

Comment:
Suggest simplifying by overriding the internal `_transferRemote(...)` function (with 5 params and add this check.

```solidity
    function _transferRemote(
        uint32 _destination,
        bytes32 _recipient,
        uint256 _amountOrId,
        uint256 _value,
        bytes memory _hookMetadata,
        address _hook
    ) internal virtual returns (bytes32 messageId) {
        if (_recipient == routers(_destination)) revert InvalidRecipientError();
        super._transferRemote(_destination, _recipient, _amountOrId, _value, _hookMetadata);
    }
```

![image](https://github.com/user-attachments/assets/aa26d0db-25dc-4b79-866c-aee51a8c7975)